roles/federation-demo/tasks/deploy.yml

```
@@ -0,0 +1,83 @@
+---
+  - name: Check if project exists
+    stat:
+      path: "{{ federation_git.clone_dest }}"
+    register: project_stat
+
+  - name: Clean clone directory
+    file:
+      state: absent
+      path: "{{ federation_git.clone_dest }}"
+    when: project_stat.stat.exists
+
+  - name: Clone the repository
+    git:
+      repo: "{{ federation_git.git_repo }}"
+      version: "{{federation_git.git_branch}}"
+      dest: "{{federation_git.clone_dest}}"
+      key_file: "{{ federation_git.key_path }}"
+
+
+  - name: Install node modules for react
+    npm:
+      path: "{{federation_react_path}}"
+
+  - name: Install node modules for express
+    npm:
+      path: "{{federation_express_path}}"
+
+  - name: Install node modules for ams-agent
+    npm:
+      path: "{{federation_ams_agent_path}}"
+
+  - name: React config
+    template:
+      src: react-config.j2
+      dest: "{{federation_react_path}}/src/config.json"
+
+  - name: Configure express db_config
+    template:
+      src: db_config.j2
+      dest: "{{federation_express_path}}/db-config/db-config.json"
+
+  - name: Expres env
+    template:
+      src: express-env.j2
+      dest: "{{federation_express_path}}/JavaScript/.env"
+
+  - name: Agent env
+    template:
+      src: ams-env.j2
+      dest: "{{federation_ams_agent_path}}/.env"
+
+  - name: delete existing pm2 processes if running
+    command: "pm2 delete all"
+    ignore_errors: True
+
+  - name: Create pm2 ecosystem
+    template:
+     src: pm2_config.j2
+     dest: "{{federation_pm2_conf_path}}/ecosystem.config.js"
+
+  - name: start Pm2 processes
+    command: "pm2 start ecosystem.config.js"
+
+  - name: restart Pm2 processes
+    command: "pm2 restart all"
+
+  - name: react build
+    command: "npm --prefix {{federation_react_path}} run build"
```

I suggest you the following:
```suggestion
    shell: "npm run build"
      args:
        chdir: "{{ federation_react_path }}"
        executable: /bin/bash
```

* https://docs.ansible.com/ansible/2.9/modules/shell_module.html#parameter-chdir
* https://docs.ansible.com/ansible/2.9/modules/command_module.html#parameter-chdir